sdk/src/types/block/semantic/mod.rs

```diff
@@ -125,10 +131,36 @@ impl<'a> SemanticValidationContext<'a> {
             }
         }
 
+        // Validation of inputs.
+
+        let mut has_implicit_account_creation_address = false;
+
         for (index, (output_id, consumed_output)) in self.inputs.iter().enumerate() {
             let (amount, consumed_native_token, unlock_conditions) = match consumed_output {
                 Output::Basic(output) => (output.amount(), output.native_token(), output.unlock_conditions()),
-                Output::Account(output) => (output.amount(), None, output.unlock_conditions()),
+                Output::Account(output) => {
+                    if output.features().block_issuer().is_some() {
+                        let account_id = output.account_id_non_null(output_id);
+
+                        if self.commitment_context_input.is_none() {
+                            return Ok(Some(TransactionFailureReason::BlockIssuerCommitmentInputMissing));
+                        }
+                        if !self.bic_context_inputs.contains(&account_id) {
+                            return Ok(Some(TransactionFailureReason::BlockIssuanceCreditInputMissing));
+                        }
+                        let entry = self.block_issuer_mana.entry(account_id).or_default();
+                        entry.0 = entry
+                            .0
+                            .checked_add(consumed_output.available_mana(
+                                &self.protocol_parameters,
+                                output_id.transaction_id().slot_index(),
+                                self.transaction.creation_slot(),
+                            )?)
+                            .ok_or(Error::ConsumedManaOverflow)?;
```

Comment:
Is this only to check the mana overflow? Isn't that also checked later again for the whole tx?

Reply:
Ah entry is mutable even with mut keyword